other/markov_chain.py

```diff
@@ -0,0 +1,80 @@
+from random import random
+from typing import Dict, List, Tuple
+
+
+class MarkovChainGraphUndirectedUnweighted:
+    '''
+    Undirected Unweighted Graph for running Markov Chain Algorithm
+    '''
+
+    def __init__(self):
+        self.connections = {}
+
+    def add_node(self, node: str) -> None:
+        self.connections[node] = {}
+
+    def add_transition_probability(self, node1: str,
+                                   node2: str,
+                                   probability: float) -> None:
+        if node1 not in self.connections:
+            self.add_node(node1)
+        if node2 not in self.connections:
+            self.add_node(node2)
+        self.connections[node1][node2] = probability
+
+    def get_nodes(self) -> List[str]:
+        return list(self.connections.keys())
```

Comment:
```suggestion
        return list(self.connections)
```
.keys() is almost never needed in modern Python.